File: playerai/src/main/java/com/example/playerai/entity/Player.java
package com.example.playerai.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "players")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Player {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private Integer age;
    private String position;
    private String team;

    private Integer matchesPlayed;
    private Integer goals;
    private Integer assists;
    private Integer minutesPlayed;
    private Integer yellowCards;
    private Integer redCards;
    private Integer shotsOnTarget;
    private Double passAccuracy;
    private Double formRating;
    private Boolean injuryStatus;
}

File: frontend/src/app/models/player.ts
export interface Player {
  id?: number;
  name: string;
  age: number;
  position: string;
  team: string;
  matchesPlayed: number;
  goals: number;
  assists: number;
  minutesPlayed: number;
  yellowCards: number;
  redCards: number;
  shotsOnTarget: number;
  passAccuracy: number;
  formRating: number;
  injuryStatus: boolean;

  expectedGoals?: number;
  expectedAssists?: number;
  keyPasses?: number;
  progressivePasses?: number;
  dribblesCompleted?: number;
  tacklesWon?: number;
  interceptions?: number;
  ballRecoveries?: number;
  matchesMissed?: number;
  recentMatchLoad?: number;
}
